### PR TITLE
Fix temb attention

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Fixes: https://github.com/huggingface/diffusers/issues/3576

### CODE REVIEW

File: src/diffusers/models/attention_processor.py [hunk 1]
```diff
@@ -834,6 +837,7 @@ def __init__(self, hidden_size, cross_attention_dim=None, rank=4):
 
     def __call__(self, attn: Attention, hidden_states, encoder_hidden_states=None, attention_mask=None, scale=1.0):
         residual = hidden_states
+
```
Comment (patrickvonplaten) [inline src/diffusers/models/attention_processor.py:840]:
```suggestion
```

**Review by sayakpaul — APPROVED** [review]
Do we need to add a test for this?

File: src/diffusers/models/attention_processor.py [hunk 2]
```diff
@@ -540,9 +540,14 @@ def __init__(self, hidden_size, cross_attention_dim=None, rank=4):
         self.to_v_lora = LoRALinearLayer(cross_attention_dim or hidden_size, hidden_size, rank)
         self.to_out_lora = LoRALinearLayer(hidden_size, hidden_size, rank)
 
-    def __call__(self, attn: Attention, hidden_states, encoder_hidden_states=None, attention_mask=None, scale=1.0):
+    def __call__(
+        self, attn: Attention, hidden_states, encoder_hidden_states=None, attention_mask=None, temb=None, scale=1.0
```
Comment (patrickvonplaten) [inline src/diffusers/models/attention_processor.py:544]:
```suggestion
        self, attn: Attention, hidden_states, encoder_hidden_states=None, attention_mask=None, scale=1.0, temb=None
```

File: docker/diffusers-pytorch-cuda/Dockerfile [hunk 3]
```diff
@@ -38,6 +38,9 @@ RUN python3 -m pip install --no-cache-dir --upgrade pip && \
         scipy \
         tensorboard \
         transformers \
-        omegaconf
+        omegaconf \
+        pytorch-lightning \
+        xformers
+
```
Comment (patrickvonplaten) [inline docker/diffusers-pytorch-cuda/Dockerfile:44]:
```suggestion
```

File: src/diffusers/utils/testing_utils.py [hunk 4]
```diff
@@ -577,3 +577,9 @@ def enable_full_determinism():
     torch.backends.cudnn.deterministic = True
     torch.backends.cudnn.benchmark = False
     torch.backends.cuda.matmul.allow_tf32 = False
+
+
+def disable_full_determinism():
+    os.environ["CUDA_LAUNCH_BLOCKING"] = "0"
```
Comment (sayakpaul) [inline src/diffusers/utils/testing_utils.py:583]: Is this for preventing memory problems?

File: tests/pipelines/stable_diffusion/test_stable_diffusion.py [hunk 5]
```diff
@@ -366,6 +368,56 @@ def test_stable_diffusion_pndm(self):
 
         assert np.abs(image_slice.flatten() - expected_slice).max() < 1e-2
 
+    @unittest.skipIf(not torch.cuda.is_available(), reason="xformers requires cuda")
+    def test_stable_diffusion_attn_processors(self):
```
Comment (sayakpaul) [inline tests/pipelines/stable_diffusion/test_stable_diffusion.py:372]: Savior!

**Pull request merged**